pkg/cmd/util/clientcmd/factory_client_access.go

```diff
@@ -0,0 +1,485 @@
+package clientcmd
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	fedclientset "k8s.io/kubernetes/federation/client/clientset_generated/federation_internalclientset"
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kclientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/typed/discovery"
+	kclientcmd "k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+	kclientcmdapi "k8s.io/kubernetes/pkg/client/unversioned/clientcmd/api"
+	"k8s.io/kubernetes/pkg/kubectl"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/homedir"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/cli/config"
+	"github.com/openshift/origin/pkg/cmd/cli/describe"
+	deployapi "github.com/openshift/origin/pkg/deploy/api"
+	deploycmd "github.com/openshift/origin/pkg/deploy/cmd"
+	imageutil "github.com/openshift/origin/pkg/image/util"
+	routegen "github.com/openshift/origin/pkg/route/generator"
+)
+
+type ring0Factory struct {
+	clientConfig kclientcmd.ClientConfig
+	// TODO: is it needed?
+	OpenShiftClientConfig kclientcmd.ClientConfig
+	discoveryFactory      kcmdutil.DiscoveryClientFactory
+	// TODO: can we replace this with the upstream version?
+	clientCache             *clientCache
+	ImageResolutionOptions  FlagBinder
+	kubeClientAccessFactory kcmdutil.ClientAccessFactory
+}
+
+func NewClientAccessFactory(optionalClientConfig kclientcmd.ClientConfig) kcmdutil.ClientAccessFactory {
```

Comment:
Need an interface here that exposes a way to get `/oapi` RESTClients and restclient.Configs right?